src/charm.py

```diff
@@ -54,6 +57,11 @@ def __init__(self, *args) -> None:
             self.on.slurm_controller_relation_changed, self._slurm_controller_relation_changed
         )
 
+        self.service_init_sync("auth-munge", False)
+        self.service_init_sync("slurm-compute", False)
+        self.service_init_sync("slurm-controller", False)
```

Comment:
Do you want to track these?

Reply:
To be honest, I am not sure. These relations are needed to make the server and client work correctly even after the subordinate relation is ready. From what I gather from syncs, they are supposed to tell you when relations are up to date, but that might not be the case.

Perhaps `stale` is the better option to use?

Reply:
For the `slurm-compute`, I don't see having a sync being useful.

On the client side, these would indicate if the local controller config and munge key were loaded. Although the relation is involved in updating these syncs, they are not about the relation but the data being pulled from the relation and stored. So, in the relation-changed handlers, the sync would be set to `False`, and no actual update of the config just then.

At some time later, when the admin allows for the daemon to be restarted, the config and key can be pulled and stored locally and the syncs updated to `True`.

The stale setting would be important here, but let me get back to you on it. I need to check something first.